### PR TITLE
[Feat.] DMS: Add new resource opentelekomcloud_dms_topic_v2

### PR DESCRIPTION
**Pull request by artem-lifshits**
## Summary of the Pull Request
New resource `opentelekomcloud_dms_topic_v2` introduced.
`opentelekomcloud_dms_topic_v1` is deprecated due to api going out-of-date. 

## PR Checklist

* [x] Refers to: #2741
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccDmsTopicsV2_basic
--- PASS: TestAccDmsTopicsV2_basic (647.36s)
PASS

Process finished with the exit code 0

=== RUN   TestAccDmsTopicsV1_basic
--- PASS: TestAccDmsTopicsV1_basic (649.04s)
PASS

Process finished with the exit code 0

```

### CODE REVIEW

**Review by muneeb-jan — DISMISSED** [review]

File: docs/resources/dms_topic_v2.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  sync_replication = true
+  retention_time   = 80
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `instance_id` - (Required) Indicates the ID of primary DMS instance.
```
Comment (anton-sidelnikov) [inline docs/resources/dms_topic_v2.md]: please add type and forcenew details

File: docs/resources/dms_topic_v2.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+## Attributes Reference
+
+All above argument parameters can be exported as attribute parameters along with attribute reference.
+
+* `size` - The partition size of the topic.
+
+* `remain_partitions` - Number of remaining partitions.
+
+* `max_partitions` - Total partitions number.
```
Comment (anton-sidelnikov) [inline docs/resources/dms_topic_v2.md:100]: import?

File: opentelekomcloud/acceptance/dms/resource_opentelekomcloud_dms_topic_v2_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+const resourceTopicV2Name = "opentelekomcloud_dms_topic_v2.topic_1"
+
+func TestAccDmsTopicsV2_basic(t *testing.T) {
+	var instanceName = fmt.Sprintf("dms_instance_%s", acctest.RandString(5))
+	var topicName = fmt.Sprintf("topic_instance_%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { common.TestAccPreCheck(t) },
+		ProviderFactories: common.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckDmsV1InstanceDestroy,
```
Comment (anton-sidelnikov) [inline opentelekomcloud/acceptance/dms/resource_opentelekomcloud_dms_topic_v2_test.go:59]: please use InitResourceCheck

File: opentelekomcloud/services/dms/common.go [hunk 4]
```diff
@@ -10,6 +10,7 @@ const (
 	errCreationClient   = "error creating OpenTelekomCloud DMSv1 client: %w"
 	errCreationClientV2 = "error creating OpenTelekomCloud DMSv2 client: %w"
 	dmsClientV2         = "dms-v2-client"
+	dmsV1Deprecated     = "This resource API is deprecated. Please use opentelekomcloud_dms_topic_v2 instead."
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/dms/common.go:13]: also add message in doc

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**